pkg/execute/notifier_test.go

```diff
@@ -0,0 +1,126 @@
+package execute
+
+import (
+	"testing"
+
+	logtest "github.com/sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/kubeshop/botkube/pkg/config"
+)
+
+func TestNotifierExecutor_Do_Success(t *testing.T) {
+	// given
+	log, _ := logtest.NewNullLogger()
+	platform := config.SlackCommPlatformIntegration
+	clusterName := "cluster-name"
+	statusArgs := []string{"notifier", "status"}
+
+	testCases := []struct {
+		Name                 string
+		InputArgs            []string
+		InputNotifierHandler NotifierHandler
+		ExpectedResult       string
+		ExpectedStatusAfter  string
+	}{
+		{
+			Name:                 "Start",
+			InputArgs:            []string{"notifier", "start"},
+			InputNotifierHandler: &fakeNotifierHandler{enabled: false},
+			ExpectedResult:       `Brace yourselves, incoming notifications from cluster "cluster-name".`,
+			ExpectedStatusAfter:  `Notifications from cluster "cluster-name" are enabled.`,
+		},
+		{
+			Name:                 "Stop",
+			InputArgs:            []string{"notifier", "stop"},
+			InputNotifierHandler: &fakeNotifierHandler{enabled: true},
+			ExpectedResult:       `Sure! I won't send you notifications from cluster "cluster-name" anymore.`,
+			ExpectedStatusAfter:  `Notifications from cluster "cluster-name" are disabled.`,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.Name, func(t *testing.T) {
+			e := NewNotifierExecutor(log, config.Config{}, &fakeAnalyticsReporter{})
+
+			// execute command
+
+			// when
+			actual, err := e.Do(tc.InputArgs, platform, clusterName, tc.InputNotifierHandler)
+
+			// then
+			require.Nil(t, err)
+			assert.Equal(t, tc.ExpectedResult, actual)
+
+			// get status after executing a given command
+
+			// when
+			actual, err = e.Do(statusArgs, platform, clusterName, tc.InputNotifierHandler)
+			// then
+			require.Nil(t, err)
+			assert.Equal(t, tc.ExpectedStatusAfter, actual)
+		})
+	}
+}
+
+func TestNotifierExecutor_Do_Error(t *testing.T) {
```

Comment:
Interesting use of underscores in the test name. [I've seen it here before](https://github.com/hashicorp/consul/blob/main/api/api_test.go#L424) (format: `Test + package name + _ + func under test + _ + outcome`)

Only curious, why not just use: `TestNotifierExecutorDoSuccess` or `TestNotifierExecutorDoError`?

Reply:
True, usually I write success/failure test cases within the same `Test` func, using the same table tests with different params. However, here, if I would merge the success/failure cases, I thought it would complicate the code and make the tests harder to maintain. But guess what? 😄 I refactored that after your comment and that doesn't look that complicated, so will merge the separate functions 👍 Thanks!